update-doi-status/src/test/java/no/unit/nva/doi/handler/UpdateDoiStatusHandlerTest.java

```diff
@@ -0,0 +1,166 @@
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import com.amazonaws.services.lambda.runtime.Context;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.exc.MismatchedInputException;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.net.URI;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.util.UUID;
+import no.unit.nva.model.Publication;
+import no.unit.nva.model.Publication.Builder;
+import no.unit.nva.publication.doi.dto.PublicationDynamoEventTestDataGenerator;
+import no.unit.nva.publication.service.PublicationService;
+import nva.commons.exceptions.ApiGatewayException;
+import nva.commons.exceptions.ApiIoException;
+import nva.commons.handlers.GatewayResponse;
+import nva.commons.utils.Environment;
+import nva.commons.utils.IoUtils;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.zalando.problem.Problem;
+
+class UpdateDoiStatusHandlerTest {
+
+    public static final Path BAD_EVENT_WITH_DATE_IN_FUTURE = Path.of(
+        "update_doi_status_event_bad_date_in_the_future.json");
+    public static final String NULL_OBJECT = "{ }";
+    public static final URI EXAMPLE_DOI = URI.create("https://doi.org/10.1103/physrevd.100.085005");
+    public static final Instant EXAMPLE_DOI_MODIFIED_DATE = Instant.parse("2020-08-14T10:30:10.019991Z");
+    private static final UUID EXAMPLE_PUBLICATION_ID = UUID.fromString("41076d56-2839-11eb-b644-1bb5be85c01b");
+    private static final Path BAD_EVENT_WITH_BAD_PAYLOAD_NOT_MATCHING_POJO = Path.of(
+        "update_doi_status_event_bad_input_not_matching_pojo.json");
+    private static final Path OK_EVENT = Path.of("update_doi_status_event.json");
+    private UpdateDoiStatusHandler handler;
+    private ByteArrayOutputStream outputStream;
+    private Context context;
+    private PublicationService publicationService;
+    private Environment environment;
+    private PublicationDynamoEventTestDataGenerator publicationGenerator;
+
+    @BeforeEach
+    void setUp() throws ApiGatewayException {
+        publicationService = mock(PublicationService.class);
+        environment = mock(Environment.class);
+        handler = new UpdateDoiStatusHandler(publicationService, environment);
+        context = mock(Context.class);
+        outputStream = new ByteArrayOutputStream();
+
+        publicationGenerator = new PublicationDynamoEventTestDataGenerator();
```

Comment:
Replace with https://github.com/BIBSYSDEV/nva-publication-api/pull/98

Reply:
We need to cycle back to this later, we need test generators who makes using random test data a joy to work with.